src/python/pants/bin/daemon_pants_runner.py

```diff
@@ -283,7 +284,13 @@ def post_fork_child(self):
     # terminal window.
     # TODO: test the above!
     ExceptionSink.reset_exiter(self._exiter)
-    ExceptionSink.reset_interactive_output_stream(sys.stderr)
+
+    try:
```

Comment:
I think that rather than doing this outside of `ExceptionSink`, you should `try:except:` inside of `reset_interactive_output_stream`, and have it use `ExceptionSink.log_exception` to render the failure.